server/converters/cxgtool.py

```diff
@@ -409,6 +442,17 @@ def save_metadata(container, metadata):
     with tiledb.DenseArray(a_name, mode="w") as A:
         A.meta["cxg_version"] = CXG_VERSION
         A.meta["cxg_properties"] = json.dumps(metadata)
+        if extract_colors:
+            try:
+                A.meta["cxg_category_colors"] = json.dumps(
+                    convert_anndata_category_colors_to_cxg_category_colors(adata)
```

Comment:
IMHO, it would be cleaner if you did all this color extraction in write_cxg(), and just provided save_metadata() with a color dict to save. That fits better with the existing API signature.

It also lets you extract all colors _before_ you start creating tiledb objects, which keeps the file system clean in the case of an error.   Ie, convert_anndata...() should be run at the beginning of write_cxg() before the initial call to tiledb.group_create().

Reply:
True. Will separate the concerns here.